### ISSUE TITLE
Oracle JDK download is broken in oracle-java8-installer installation

### ISSUE THREAD

**Issue by mshuler** [datastax/docker-images#17]
Switch to OpenJDK or switch to a functioning upstream docker JDK8 image parent?
```
<...>
DEBUG BuildResponseItem[stream=Setting up java-common (0.56ubuntu2) ...
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=Setting up locales (2.23-0ubuntu10) ...
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=Generating locales (this might take a while)...
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=Generation complete.
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=Setting up oracle-java8-installer (8u151-1~webupd8~0) ...
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=No /var/cache/oracle-jdk8-installer/wgetrc file found.
Creating /var/cache/oracle-jdk8-installer/wgetrc and
using default oracle-java8-installer wgetrc settings for it.
Downloading Oracle Java 8...
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=--2018-01-17 16:12:33--  http://download.oracle.com/otn-pub/java/jdk/8u151-b12/e758a0de34e24606bca991d704f6dcbf/jdk-8u151-linux-x64.tar.gz
Resolving download.oracle.com (download.oracle.com)... ,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=23.64.147.55
Connecting to download.oracle.com (download.oracle.com)|23.64.147.55|:80... ,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=failed: Connection timed out.
Retrying.

,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=--2018-01-17 16:14:44--  (try: 2)  http://download.oracle.com/otn-pub/java/jdk/8u151-b12/e758a0de34e24606bca991d704f6dcbf/jdk-8u151-linux-x64.tar.gz
Connecting to download.oracle.com (download.oracle.com)|23.64.147.55|:80... ,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=failed: Connection timed out.
Retrying.

,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=--2018-01-17 16:16:56--  (try: 3)  http://download.oracle.com/otn-pub/java/jdk/8u151-b12/e758a0de34e24606bca991d704f6dcbf/jdk-8u151-linux-x64.tar.gz
Connecting to download.oracle.com (download.oracle.com)|23.64.147.55|:80... ,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=failed: Connection timed out.
Retrying.

,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=--2018-01-17 16:19:10--  (try: 4)  http://download.oracle.com/otn-pub/java/jdk/8u151-b12/e758a0de34e24606bca991d704f6dcbf/jdk-8u151-linux-x64.tar.gz
Connecting to download.oracle.com (download.oracle.com)|23.64.147.55|:80... ,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=failed: Connection timed out.
Retrying.

,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=--2018-01-17 16:21:24--  (try: 5)  http://download.oracle.com/otn-pub/java/jdk/8u151-b12/e758a0de34e24606bca991d704f6dcbf/jdk-8u151-linux-x64.tar.gz
Connecting to download.oracle.com (download.oracle.com)|23.64.147.55|:80... ,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=failed: Connection timed out.
Giving up.

,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=download failed
Oracle JDK 8 is NOT installed.
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=dpkg: error processing package oracle-java8-installer (--configure):
 subprocess installed post-installation script returned error exit status 1
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=Errors were encountered while processing:
 oracle-java8-installer
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=E: Sub-process /usr/bin/dpkg returned an error code (1)
,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=<null>,error=<null>,aux=<null>]
DEBUG BuildResponseItem[stream=<null>,status=<null>,progressDetail=<null>,progress=<null>,id=<null>,from=<null>,time=<null>,errorDetail=ResponseItem.ErrorDetail[code=100,message=The command '/bin/sh -c apt-get update &&     apt-get install -y software-properties-common &&     add-apt-repository ppa:webupd8team/java -y &&     apt-get update &&     echo debconf shared/accepted-oracle-license-v1-1 select true |  debconf-set-selections &&     echo debconf shared/accepted-oracle-license-v1-1 seen true |  debconf-set-selections &&     apt-get install  -y --no-install-recommends ca-certificates &&     apt-get -y dist-upgrade &&     apt-get install -y --no-install-recommends oracle-java8-installer &&     apt-get install -y --no-install-recommends oracle-java8-set-default &&     apt-get autoremove -y &&     rm -rf /var/cache/oracle-jdk8-installer &&     ln -s java-8-oracle  /usr/lib/jvm/default-jvm &&     apt-get clean all' returned a non-zero code: 100],error=The command '/bin/sh -c apt-get update &&     apt-get install -y software-properties-common &&     add-apt-repository ppa:webupd8team/java -y &&     apt-get update &&     echo debconf shared/accepted-oracle-license-v1-1 select true |  debconf-set-selections &&     echo debconf shared/accepted-oracle-license-v1-1 seen true |  debconf-set-selections &&     apt-get install  -y --no-install-recommends ca-certificates &&     apt-get -y dist-upgrade &&     apt-get install -y --no-install-recommends oracle-java8-installer &&     apt-get install -y --no-install-recommends oracle-java8-set-default &&     apt-get autoremove -y &&     rm -rf /var/cache/oracle-jdk8-installer &&     ln -s java-8-oracle  /usr/lib/jvm/default-jvm &&     apt-get clean all' returned a non-zero code: 100,aux=<null>]

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildDseBaseImage'.
> Docker command failed: Could not build image: The command '/bin/sh -c apt-get update &&     apt-get install -y software-properties-common &&     add-apt-repository ppa:webupd8team/java -y &&     apt-get update &&     echo debconf shared/accepted-oracle-license-v1-1 select true |  debconf-set-selections &&     echo debconf shared/accepted-oracle-license-v1-1 seen true |  debconf-set-selections &&     apt-get install  -y --no-install-recommends ca-certificates &&     apt-get -y dist-upgrade &&     apt-get install -y --no-install-recommends oracle-java8-installer &&     apt-get install -y --no-install-recommends oracle-java8-set-default &&     apt-get autoremove -y &&     rm -rf /var/cache/oracle-jdk8-installer &&     ln -s java-8-oracle  /usr/lib/jvm/default-jvm &&     apt-get clean all' returned a non-zero code: 100

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org
```

**Comment by mshuler** [comment]
Oracle just released jdk8u161 & u162, so the failure makes sense. The fix will have to wait for someone to commit the change to the Ubuntu PPA for the installer package, but changing to OpenJDK should always work, since it's a binary package in the OS repository instead of some fragile downloader.

**Comment by gmflau** [comment]
DataStax had released DSE 5.1.6 last week to run on top of JDK 8u162.  Can you pls bump up the version of DSE to 5.1.6 in the Dockerfile - https://github.com/datastax/docker-images/blob/master/server/5.1/Dockerfile#L9 ?

**Comment by roberd13** [comment]
5.1.6 tag has been added to github Dockerfile and pushed to docker store which resolves the 161 and 162 issues.  Also instead of pulling upstream, We have changed base to pull from our artifactory instead.
